lib/core/utils/publish-metadata.js

```diff
@@ -49,18 +13,12 @@ function extender(checksData, shouldBeTrue) {
 		var data = Object.assign({}, sourceData);
 		delete data.messages;
 		if (check.result === undefined) {
-			if (typeof messages.incomplete === 'object') {
-				data.message = function() {
-					return getIncompleteReason(check.data, messages);
-				};
-			} else {
-				// fall back to string function
-				data.message = messages.incomplete;
-			}
+			data.message = messages.incomplete;
 		} else {
 			data.message =
 				check.result === shouldBeTrue ? messages.pass : messages.fail;
 		}
+		data.message = axe.utils.processMessage(data.message, check.data);
```

Comment:
This isn't going to work if someone passes in doT.js messages into `axe.configure()`. We need to support both message systems, otherwise this would be a breaking change.